flake.nix

```diff
@@ -57,12 +64,34 @@
         in {
           inherit (localPkgs) prep-unison-scratch;
 
+          emacsPackagesFor = emacs:
+            (prev.emacsPackagesFor emacs).overrideScope'
+            (self.overlays.emacs final prev);
+
+          tree-sitter = prev.tree-sitter.override {
+            extraGrammars = self.lib.tree-sitter-grammars final;
+          };
+
           ## Renamed to replace the `unison-ucm` included in Nixpkgs.
           unison-ucm = localPkgs.ucm;
 
           vimPlugins = prev.vimPlugins // self.overlays.vim final prev;
         };
 
+        emacs = final: prev: efinal: eprev: {
+          unison-ts-mode = efinal.trivialBuild {
+            pname = "unison-ts-mode";
+            version = "1.0.0-rc.1";
+
+            src = final.fetchFromGitHub {
+              owner = "fmguerreiro";
+              repo = "unison-ts-mode";
+              rev = "main";
```

Comment:
Shouldn't this `rev` be a tag or commit hash? I'd expect sha256 mismatches if fmguerreiro pushes to `main`.

Reply:
Whoops. Fixed and pushed.